events/importer/enkora.py

```diff
@@ -1099,35 +1450,86 @@ def _handle_course(self, course: dict) -> dict:
             "audience_min_age": audience_min_age,
             "audience_max_age": audience_max_age,
             "audience": audience_keywords,
+            "event_status": Event.Status.SCHEDULED,
             "start_time": dates["first_event_date"],
+            "has_start_time": True,
             "end_time": dates["last_event_date"],
+            "has_end_time": True,
             "date_published": dates["public_visibility_start"],
-            "external_links": {"fi": {"registration": "https://www.hel.fi/fi"}},
+            "external_links": {"fi": {"registration": liikuntakauppa_fi_link}},
             "provider": {"fi": self.PROVIDER},
-            "provider_contact_info": {"fi": "Hki - KUVA - Liikunta"},
+            "provider_contact_info": {
+                "fi": "Helsingin Kaupunki - Liikuntaluuri, Avoinna ma-to 13-15"
+            },
             "enrolment_start_time": dates["public_reservation_start"],
             "enrolment_end_time": dates["public_reservation_end"],
             "maximum_attendee_capacity": capacity,
             "remaining_attendee_capacity": remaining_capacity,
             "data_source": self.data_source,
             "origin_id": course["reservation_event_group_id"],
             "publisher": self.organization,
-            "location": location_id,
+            "location": location_data,
+            "location_extra_info": location_extra_info,
             "keywords": keywords,
             "in_language": in_language,
+            "offers": offers,
             "images": [],  # must have a list of images, even if there are no images
-            "offers": [],  # must have a list of offers, even if there are no offers
         }
 
-        if location_extra_info:
-            event_data["location_extra_info"] = {"fi": location_extra_info}
+        # Sub-events:
+        # Literally all Enkora courses are multi-event
+        _, sub_events = self.build_sub_events(course, event_data)
+
+        #
```

Comment:
You could be a bit more verbose with this comment. :wink: Why we want to print this?

Reply:
Print?
The comment is about the fact, there are no single-event courses in Enkora. They do not exist. Ever. Every single course since dawn of time has had a minimum of two sub-events in it. This may not stand true for some other event source.

Reply:
I mean that you have an empty comment here `#` either remove it or write something to it.

Reply:
Ok. It's a typo and needs to go away.

Reply:
Improved comment on cache expiry.